fastseq/optimizer/fairseq/el_attention_optimizer.py

```diff
@@ -0,0 +1,1459 @@
+
+"""Apply the beam search optimizations to fairseq-v0.9.0"""
+
+import math
+from typing import Optional
+from typing import List
+from collections import namedtuple
+import torch
+import torch.nn.functional as F
+from torch import Tensor
+import random
+from fairseq import utils
+from fairseq.models.transformer import TransformerEncoder, TransformerModel
+from fairseq.modules.multihead_attention import MultiheadAttention
+from fairseq.sequence_generator import SequenceGenerator
+from fairseq.models.fairseq_model import FairseqEncoderDecoderModel
+from fairseq.models.transformer import TransformerAlignModel
+from fairseq.models.transformer import TransformerEncoder
+from fairseq.models.transformer import TransformerDecoder
+from fairseq.modules.transformer_layer import TransformerDecoderLayer
+from fairseq.sequence_generator import EnsembleModel
+from fairseq.tasks.fairseq_task import FairseqTask
+from fairseq.data.data_utils import collate_tokens
+from fastseq.utils.api_decorator import replace
+from fastseq.ops.ngram_repeat_block import NGramRepeatBlock
+from fastseq import config
+
+
+#Efficient-Lossless Attention
+use_el_attn = config.USE_EL_ATTN == '1'
```

Comment:
After converting `USE_EL_ATTN` to boolean type in config.py, `USE_EL_ATTN` can be directly used here.

Reply:
done